posthog/api/test/test_user.py

```diff
@@ -210,6 +218,15 @@ def test_update_current_user(self, mock_capture):
             groups={"instance": ANY, "organization": str(self.team.organization_id), "project": str(self.team.uuid),},
         )
 
+    def test_cannot_upgrade_yourself_to_staff_user(self):
```

Comment:
I also suggest testing that it's not possible to _demote_ yourself, as doing that accidentally would force the admin to connect to the DB directly to fix it, which could end up as an extra support inquiry for us.

Reply:
I think this is fine for now, this is intended for advanced users who are likely to know how to break glass. Further, I do see legitimate case for wanting to demote yourself (e.g. if you're in the process of deactivating your account). I'm in favor of flexibility here when it's intended for advanced users.